src/Scout/ScoutEnhancer.php

```diff
@@ -64,19 +64,25 @@ public function wouldEnhanceBuilder(): bool
         return $this->hasSearchArguments();
     }
 
-    public function enhanceBuilder(): ScoutBuilder
+    /** @param (callable(\Nuwave\Lighthouse\Scout\ScoutBuilderDirective): bool)|null $directiveFilter */
+    public function enhanceBuilder(callable $directiveFilter = null): ScoutBuilder
     {
         $scoutBuilder = $this->builder instanceof ScoutBuilder
             ? $this->builder
             : $this->enhanceEloquentBuilder();
 
         foreach ($this->argumentsWithScoutBuilderDirectives as $argument) {
-            $scoutBuilderDirective = $argument
-                ->directives
-                ->first(Utils::instanceofMatcher(ScoutBuilderDirective::class));
-            assert($scoutBuilderDirective instanceof ScoutBuilderDirective);
+            foreach ($argument->directives as $directive) {
```

Comment:
Writing this as `foreach` is fine, please rewrite `ResolveInfo::applyArgBuilderDirectives()` in the same way.